lib/Folder.php

```diff
@@ -62,6 +68,9 @@ public function __construct(int $accountId, Horde_Imap_Client_Mailbox $mailbox,
 		$this->folders = [];
 		$this->status = [];
 		$this->specialUse = [];
+		$lastSeparator = strrpos($mailbox->utf8, $delimiter) > 0 ? strrpos($mailbox->utf8, $delimiter)+1: 0;
+		$this->displayName = substr($mailbox->utf8, $lastSeparator);
+		$this->path = substr($mailbox->utf8, 0, $lastSeparator);
 	}
```

Comment:
Seperate display name and path. The path is used to select the subfolders.

Reply:
we only need it on the front-end. can we just generate this there?

Reply:
Yes, we can;-), separating path and file name and creating the hierarchy on the fly is ok so far?

Reply:
I'm not entirely sure. But I did not fully grasp the intended change so far